gradio/documentation.py

```diff
@@ -0,0 +1,146 @@
+import inspect
+
+classes_to_document = {}
+gl = {"mode": None}
+
+
+def document_mode(m):
+    gl["mode"] = m
+    if m not in classes_to_document:
+        classes_to_document[m] = []
+
+
+def document(*fns):
+    def inner_doc(cls):
+        classes_to_document[gl["mode"]].append((cls, fns))
+        return cls
+
+    return inner_doc
+
+
+def document_fn(fn):
+    doc_str = inspect.getdoc(fn)
+    doc_lines = doc_str.split("\n")
+    signature = inspect.signature(fn)
+    description, parameters, returns, examples = [], {}, [], []
+    mode = "description"
+    for line in doc_lines:
+        if line == "Parameters:":
+            mode = "parameter"
+        elif line == "Example:":
+            mode = "example"
+        elif line == "Returns:":
+            mode = "return"
+        else:
+            if mode == "description":
+                description.append(line if line != "" else "<br>")
+                continue
+            if line.startswith("    "):
+                line = line[4:]
+            if mode == "parameter":
+                colon_index = line.index(": ")
```

Comment:
Use `line.split(":")`

Reply:
current way is more robust in case there's a second colon in the line somewhere.